### ISSUE TITLE
Series of far future dates `max()` error - windows only

### ISSUE THREAD

**Issue by oscar6echo** [pola-rs/polars#4258]
#### What language are you using?

**Python**.

#### Have you tried latest version of polars?

- yes

#### What version of polars are you using?

0.13.60

#### What operating system are you using polars on?

Windows 10.
The bug does **NOT** show on Linux.

#### What language version are you using

Python 3.8

#### Describe your bug.

With far future dates e.g. 9009, `Series.max()` yields an error.  
None of us will live that long, I agree, but it is still a strange bug.  

#### What are the steps to reproduce the behavior?

```py
import datetime as dt
import polars as pl

# cell 1
dates  = [
    dt.date(2022, 8, 2),
    dt.date(2096, 8, 1),
    dt.date(9009, 9, 9),
]
se = pl.Series(dates, dtype=pl.Date)
print(se)
# output
shape: (3,)
Series: '' [date]
[
	2022-08-02
	2096-08-01
	9009-09-09
]

# cell 2
se.max()

# output linux
datetime.date(9009, 9, 9)

# error windows
---------------------------------------------------------------------------
PanicException                            Traceback (most recent call last)
Input In [30], in <cell line: 1>()
----> 1 se.max()

File C:\HOMEWARE\Miniconda3-Windows-x86_64\envs\test\lib\site-packages\polars\internals\series.py:900, in Series.max(self)

    889 def max(self) -> int | float | date | datetime | timedelta:
    890     """
    891     Get the maximum value in this Series.
    892
   (...)
    898
    899     """
--> 900     return self._s.max()

PanicException: called `Result::unwrap()` on an `Err` value: PyErr { type: <class 'OSError'>, value: OSError(22, 'Invalid argument'), traceback: Some(<traceback object at 0x0000025215A22480>) }
```

#### What is the expected behavior?

Windows output should be like linux output.

**Comment by alexander-beedie** [comment]
@oscar6echo - @gaoxinge fixed this issue, and polars 0.14.0 should have the updated code; could you try it out to confirm, and close this issue if it works? Thanks :)

**Comment by oscar6echo** [comment]
@alexander-beedie Sorry I've been on holidays traveling. Could not test.  
@gaoxinge Thx for the fix :+1:
